packages/form/src/components/File/hooks.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback } from "react";
+
+import type { FileExtended, FileMode } from "./types";
+
+export const useOnDropFile = ({
+  mode,
+  multiple,
+  name,
+  onChange,
+  value,
+}: {
+  name: string;
+  multiple: boolean;
+  mode: FileMode;
+  value: FileExtended[];
+  onChange: (file: FileExtended[]) => void;
+}) =>
+  useCallback(
+    (droppedFiles: FileExtended[]) => {
+      /*
+         This is where the magic is happening.
+         Depending upon the mode we are replacing old files with new one,
+         or appending new files into the old ones, and also filtering out the duplicate files. 
+      */
+      let newFiles =
+        mode === "update" || !multiple
+          ? droppedFiles
+          : [...(value || []), ...droppedFiles];
+
+      newFiles = newFiles.reduce((previous: FileExtended[], file) => {
+        const fo = Object.entries(file);
+
+        if (
+          previous.find((pFile: FileExtended) => {
+            const eo = Object.entries(pFile);
+            return eo.every(
+              ([key, value], index) =>
+                key === fo[index][0] && value === fo[index][1],
+            );
+          })
+        ) {
+          return previous;
+        } else {
+          return [...previous, file];
+        }
```

Comment:
We do this only if mode is "append"? I think we can refactor this to increase readibility.

Reply:
I have test the code, it is required for both other wise same file can selected again and again, in short problem of duplication occurs

I case of readability, refactor will help.